dali/pipeline/data/tensor_list.cc

```diff
@@ -791,11 +791,16 @@ void TensorList<Backend>::SetContiguity(BatchContiguity state) {
 
 
 template <typename Backend>
-void TensorList<Backend>::MakeContiguous(std::weak_ptr<void> owner) {
+void TensorList<Backend>::MakeContiguous(AccessOrder copy_order) {
   if (state_.IsContiguous()) {
     return;
   }
-  DALI_FAIL("Coalescing the buffer to Contiguous state is not yet implemented.");
+  auto contiguous_tmp = TensorList<Backend>(shape().num_samples());
+  contiguous_tmp.SetContiguity(BatchContiguity::Contiguous);
+  contiguous_tmp.SetupLike(*this);
+  contiguous_tmp.Copy(*this, order());
+  copy_order.wait(order());
```

Comment:
I don't remember how it works exactly, but probably this copy_order can be used as parameter for Copy.

Reply:
I think both ways would work, I don't know which is better.